std/concurrency.d

```diff
@@ -2397,12 +2397,12 @@ version (unittest)
 private @property Mutex initOnceLock()
 {
     __gshared Mutex lock;
```

Comment:
> We need to fix initOnce so that it takes a shared(Mutex) and deprecate the overload that takes a non-shared one.

As far as I see, that's a separate issue. It can be fixed independently from issue 16230.

I suggest to fix it afterwards. Fixing it here would mean either
1) making the phobos and druntime PRs mutually dependent (not fun, in my experience), or
2) adding casts to shared that won't be needed once issue 16230 is fixed.

And I'd rather not add another deprecation discussion to this, seeing that the druntime PR has been stuck in one for a long time.

I think it's going to be a smoother ride when we do this in three steps. Here's a PR for the third step (to be pulled after the druntime PR): #5661.